node-gui-x/src-tauri/src/lib.rs

```diff
@@ -0,0 +1,1045 @@
+                                }
+                            }
+                        }
+                    }
+
+                    Some(BackendEvent::ToggleStaking(msg)) => {
+                        match msg {
+                            Ok(transaction_info) => {
+                                if let Some(app_handle) = GLOBAL_APP_HANDLE.get() {
+                                    app_handle.emit("ToggleStaking", transaction_info).unwrap();
+                                }
+                            }
+                            Err(e) => {
+                                let error_message = e.to_string();
+                                if let Some(app_handle) = GLOBAL_APP_HANDLE.get() {
+                                    app_handle.emit("Error", error_message).unwrap();
+                                }
+                            }
+                        }
+                    }
+
+                    Some(BackendEvent::ConsoleResponse(wallet_id, account_id, result)) => {
+                        match result {
+                            Ok(console_command) => {
+                                    if let Some(app_handle) = GLOBAL_APP_HANDLE.get() {
+                                    app_handle.emit("ConsoleResponse", console_command).unwrap();
+                                }
+                            }
+                            Err(e) => {
+                                let error_message = e.to_string();
+                                if let Some(app_handle) = GLOBAL_APP_HANDLE.get() {
+                                    app_handle.emit("ConsoleResponse", error_message).unwrap();
+                                }
+                            }
+                        }
+                    }
+
+                    Some(BackendEvent::Broadcast(msg)) => {
+                        match msg {
+                            Ok(wallet_id) => {
+                                if let Some(app_handle) = GLOBAL_APP_HANDLE.get() {
+                                    app_handle.emit("Broadcast", wallet_id).unwrap();
+                                }
+                            }
+                            Err(e) => {
+                                let error_message = e.to_string();
+                                if let Some(app_handle) = GLOBAL_APP_HANDLE.get() {
+                                    app_handle.emit("Error", error_message).unwrap();
+                                }
+                            }
+                        }
+                    }
+
+                    None => {
+                        println!("No message received from backend");
+                    }
+                    _ => {
+                        println!("Received an unhandled backend event");
+                    }
+                }
+                    }
+                }
+        tokio::select! {
+                    msg_opt = low_priority_backend_receiver.recv() =>{
+        match msg_opt {
```

Comment:
same comments regarding matches applies here. General idea is that this function is too big and unreadable. Even rust formatting can't handle it.